liquibase-core/src/main/java/liquibase/database/core/SybaseDatabase.java

```diff
@@ -231,43 +237,55 @@ public boolean supportsDropTableCascadeConstraints() {
         return false;
     }
 
-	@Override
-	public String getViewDefinition(CatalogAndSchema schema, String viewName) throws DatabaseException {
+    @Override
+    public String getViewDefinition(CatalogAndSchema schema, String viewName) throws DatabaseException {
         schema = schema.customize(this);
         GetViewDefinitionStatement statement = new GetViewDefinitionStatement(schema.getCatalogName(), schema.getSchemaName(), viewName);
         Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc", this);
         @SuppressWarnings("unchecked")
         List<String> definitionRows = (List<String>) executor.queryForList(statement, String.class);
         StringBuilder definition = new StringBuilder();
         for (String d : definitionRows) {
-        	definition.append(d);
+            definition.append(d);
+        }
+        /*delete the words "CREATE VIEW [name_view] as"*/
+        String defUpper = definition.toString().toUpperCase();
+        int findPos = 0;
+        if (defUpper.contains(" AS ")) {
+            findPos = defUpper.indexOf(" AS ") + 3;
+        } else if (defUpper.contains(" AS\n")) {
+            findPos = defUpper.indexOf(" AS\n") + 3;
+        } else if (defUpper.contains("\nAS ")) {
+            findPos = defUpper.indexOf("\nAS ") + 3;
+        } else if (defUpper.contains("\nAS\n")) {
+            findPos = defUpper.indexOf("\nAS\n") + 3;
         }
-        return definition.toString();
-	}
-	
-	/** 
-	 * @return the major version if supported, otherwise -1
-	 * @see liquibase.database.AbstractJdbcDatabase#getDatabaseMajorVersion()
-	 */
-	@Override
+        return definition.substring(findPos, definition.toString().length());
+    }
+
+    /**
+     * @return the major version if supported, otherwise -1
+     * @see liquibase.database.AbstractJdbcDatabase#getDatabaseMajorVersion()
+     */
+    @Override
     public int getDatabaseMajorVersion() throws DatabaseException {
         if (getConnection() == null) {
             return -1;
         }
         try {
             return getConnection().getDatabaseMajorVersion();
         } catch (UnsupportedOperationException e) {
-        	Scope.getCurrentScope().getLog(getClass())
-        		.warning("Your JDBC driver does not support getDatabaseMajorVersion(). Consider upgrading it.");
+            Scope.getCurrentScope().getLog(getClass())
+                    .warning("Your JDBC driver does not support getDatabaseMajorVersion(). Consider upgrading it.");
```

Comment:
Is there a way to include the Sybase JDBC driver version in the error message? 

I'd also like to see the the warning message explain why it matters that the getDatabaseMajorVersion() is problematic. Perhaps something like, "Your JDBC driver [version] does not support getDatabaseMajorVersion() which can cause [some explanation of what problem this causes]"?

Reply:
I don't think the driver version is exposed in a way we can get at from here. 

This error message is for "old" drivers from 12 years ago. I'd be fine just removing that try/catch and letting the UnsupportedOperationException get thrown at this point